core/.js/src/main/scala/io/udash/component/Listenable.scala

```diff
@@ -1,27 +1,45 @@
 package io.udash.component
 
-import io.udash.utils.{CallbacksHandler, Registration}
+import com.avsystem.commons._
+import io.udash.Registration
 
-/** A component exposing events. */
-trait Listenable[ComponentType <: Listenable[ComponentType, _], EventType <: ListenableEvent[ComponentType]] {
-  private val actions = new CallbacksHandler[EventType]
+trait Listenable {
+  type EventType <: ListenableEvent
+  final type EventHandler = PartialFunction[EventType, Unit]
+
+  //visible for testing
+  private[udash] val listeners = MLinkedHashSet.empty[EventHandler]
+
+  /**
+   * Register event handler in component.
```

Comment:
Nit: I would recommend to improve API docs slightly:
1. _Registers_ and _Removes_ (line 25) should be probably better,
2. `Registration` object allows to do more with a listener than just to deregister it (line 17),
3. we could add some sort of docs to the `Listenable` and `ListenableEvent` traits, considering callbacks execution order and other general aspects.